### PR TITLE
226 Prepare project for serverless setup, describe deployment in travis

### LINKED ISSUE #226: Add staging environment
AC:
* there is staging environment built and deployed after every merge to master

### PR DESCRIPTION
**Pull request by arturtamborski**
Closes #226

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+docker build -t codeforpoznan/pah-fm-frontend:latest frontend
+docker build -t codeforpoznan/pah-fm-backend:latest  backend
+docker push codeforpoznan/pah-fm-frontend
+docker push codeforpoznan/pah-fm-backend
+
+# build and push statics
+(cd frontend && npm     run       build                    && cp -r dist   ../public)
+(cd backend  && python3 manage.py collectstatic --no-input && cp -r static ../public)
+aws s3         sync                public s3://codeforpoznan-public/dev_pah_fm
+aws cloudfront create-invalidation --paths "/*" --distribution-id E2ESZ1QPNV00X
```
Comment (arturtamborski) [inline deploy.sh:14]:
```suggestion
aws cloudfront create-invalidation --paths "/*" --distribution-id EEQ58KIXKBEQ9
```

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# bundle application
+pip install --quiet -r backend/requirements/base.txt --target packages
+(cd packages && rm -r psycopg2 && svn checkout https://github.com/jkehler/awslambda-psycopg2/trunk/psycopg2-3.6 && mv psycopg2-3.6 psycopg2)
```
Comment (arturtamborski) [inline deploy.sh:18]: it's ugly but it works, psycopg2 has some problems under the hood, that's an easy way of dealing with them

**Comment by pniedzwiedzinski** [comment]
Is `deploy.sh` used to deploy to `dev.pahfm`?

**Comment by arturtamborski** [comment]
yes, you'll also have to have few environment variables set up but yes, this script does everything that's needed to deploy the app.

**Comment by arturtamborski** [comment]
@magul / @pniedzwiedzinski can you do a review of this? I'm happy to explain every change here if it's unclear on our next hacknight, I'd love to finally get this PR merged. There's only so much I can do to test it before merging and I think that the last step here is to confirm that the deployment works for real after merging it.

**Comment by arturtamborski** [comment]
bumping @magul / @pniedzwiedzinski

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+printenv DOCKER_PASSWORD | docker login -u "$DOCKER_USERNAME" --password-stdin
+docker build -t codeforpoznan/pah-fm-frontend:latest frontend
+docker build -t codeforpoznan/pah-fm-backend:latest  backend
+docker push codeforpoznan/pah-fm-frontend
+docker push codeforpoznan/pah-fm-backend
+
+# build and push statics
+(cd frontend && npm     run       build                    && cp -r dist   ../public)
+(cd backend  && python3 manage.py collectstatic --no-input && cp -r static ../public)
+aws s3          sync    --quiet   public s3://codeforpoznan-public/dev_pah_fm
```
Comment (magul) [inline deploy.sh:13]: `---delete`

File: deploy.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+# build and push images
+printenv DOCKER_PASSWORD | docker login -u "$DOCKER_USERNAME" --password-stdin
+docker build -t codeforpoznan/pah-fm-frontend:latest frontend
+docker build -t codeforpoznan/pah-fm-backend:latest  backend
+docker push codeforpoznan/pah-fm-frontend
+docker push codeforpoznan/pah-fm-backend
+
+# build and push statics
+(cd frontend && npm     run       build                    && cp -r dist   ../public)
```
Comment (magul) [inline deploy.sh:11]: maybe `mkdir` to be more explicit how `cp` works here.

File: deploy.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+(cd backend  && python3 manage.py collectstatic --no-input && cp -r static ../public)
+aws s3          sync    --quiet   public s3://codeforpoznan-public/dev_pah_fm
+aws cloudfront create-invalidation --paths "/*" --distribution-id EEQ58KIXKBEQ9
+
+# bundle application
+pip install --quiet -r backend/requirements/base.txt --target packages
+(cd packages && rm -r psycopg2 && svn checkout https://github.com/jkehler/awslambda-psycopg2/trunk/psycopg2-3.6 && mv psycopg2-3.6 psycopg2)
```
Comment (magul) [inline deploy.sh:18]: It worked previously with migrations lambda, so we should re-check why do we need to make this hack with binaries?

**Pull request merged**